contracts/NonfungiblePositionManager.sol

```diff
@@ -106,34 +94,54 @@ contract NonfungiblePositionManager is INonfungiblePositionManager, BaseLiquidit
-    ) internal checkDeadline(deadline) returns (BalanceDelta delta, uint256 tokenId) {
-        delta = _increaseLiquidity(owner, range, liquidity, hookData, sender);
-
+        bytes memory hookData
+    ) internal checkDeadline(deadline) returns (BalanceDelta delta) {
         // mint receipt token
-        _mint(owner, (tokenId = nextTokenId++));
+        uint256 tokenId;
+        unchecked {
+            tokenId = nextTokenId++;
+        }
+        _mint(owner, tokenId);
+
+        (delta,) = _modifyLiquidity(range, liquidity.toInt256(), bytes32(tokenId), hookData);
+
         tokenPositions[tokenId] = TokenPosition({owner: owner, range: range});
     }
 
-    function increaseLiquidity(uint256 tokenId, uint256 liquidity, bytes memory hookData, bool claims, address sender)
+    // Note: Calling increase with 0 will accrue any underlying fees.
+    function increaseLiquidity(uint256 tokenId, uint256 liquidity, bytes memory hookData, address sender)
         internal
         isAuthorizedForToken(tokenId, sender)
         returns (BalanceDelta delta)
     {
         TokenPosition memory tokenPos = tokenPositions[tokenId];
-
-        delta = _increaseLiquidity(tokenPos.owner, tokenPos.range, liquidity, hookData, sender);
+        // Note: The tokenId is used as the salt for this position, so every minted liquidity has unique storage in the pool manager.
+        (delta,) = _modifyLiquidity(tokenPos.range, liquidity.toInt256(), bytes32(tokenId), hookData);
     }
 
-    function decreaseLiquidity(uint256 tokenId, uint256 liquidity, bytes memory hookData, bool claims, address sender)
+    // Note: Calling decrease with 0 will accrue any underlying fees.
+    function decreaseLiquidity(uint256 tokenId, uint256 liquidity, bytes memory hookData, address sender)
         internal
         isAuthorizedForToken(tokenId, sender)
         returns (BalanceDelta delta)
     {
         TokenPosition memory tokenPos = tokenPositions[tokenId];
+        (delta,) = _modifyLiquidity(tokenPos.range, -(liquidity.toInt256()), bytes32(tokenId), hookData);
+    }
 
-        delta = _decreaseLiquidity(tokenPos.owner, tokenPos.range, liquidity, hookData);
+    // there is no authorization scheme because the payer/recipient is always the sender
+    // TODO: Add more advanced functionality for other payers/recipients, needs auth scheme.
+    function close(Currency currency, address sender) internal returns (int256 currencyDelta) {
+        // this address has applied all deltas on behalf of the user/owner
+        // it is safe to close this entire delta because of slippage checks throughout the batched calls.
+        currencyDelta = manager.currencyDelta(address(this), currency);
+
+        // the sender is the payer or receiver
+        if (currencyDelta < 0) {
+            currency.settle(manager, sender, uint256(-int256(currencyDelta)), false);
+        } else {
+            currency.take(manager, sender, uint256(int256(currencyDelta)), false);
```

Comment:
if we're always going to pass in `false` i feel like we should use a more optimised library that doesnt need to do an `if (bool)` on every call to take